utils/genesis/src/lib.rs

```diff
@@ -161,6 +162,11 @@ where
             .tipset_by_height(0, ts.clone(), true)
             .await?;
         sm.chain_store().set_genesis(&gb.blocks()[0])?;
+        if let Some(expected_genesis_cid) = &sm.chain_config().genesis_cid {
+            if expected_genesis_cid != &gb.blocks()[0].cid().to_string() {
+                bail!("Snapshot incompatible with {}. Consider specifying the network with `--chain` flag or use a custom config file to set expected genesis CID for selected network", sm.chain_config().name);
+            }
+        }
```

Comment:
You can do some code golf to make it more concise and avoid one level of indentation. Personally I'd find this more readable but it's not a big deal, up to you.
```suggestion
        if !matches!(&sm.chain_config().genesis_cid, Some(expected) if expected ==  &gb.blocks()[0].cid().to_string())
        {
            bail!("Snapshot incompatible with {}. Consider specifying the network with `--chain` flag or use a custom config file to set expected genesis CID for selected network", sm.chain_config().name);
        }

```

Reply:
this is better, thanks @LesnyRumcajs